release.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+###############################################################################################
+### RELEASING OPERATOR-LIB
+# Every operator-lib release should have a corresponding git semantic version tag
+# begining with `v`, ex.n `v1.2.3`.
+#
+# STEP 1: Create a release branch with the name vX.Y.x. Example: git checkout -b v1.2.x
+#
+# STEP 2: Run the release script by providing the operator-lib release version as an argument
+# in the above mentioned format. Example: ./release vX.Y.Z
+#
+# STEP 3: This script will create a release tag locally. Push the release branch and tag:
+# git push upstream <release-branch>
+# git push upstream <tag-name>, wherein <tag-name> is the release version.
+#
+# STEP 4: Update the release notes in github with the changes included in corresponding
+# operator-lib version.
+#################################################################################################
+
+set -eu
+
+if [[ $# != 1 ]]; then
+	echo "usage: $0 vX.Y.Z"
+	exit 1
+fi
+
+VER=$1
+NUMRE="0|[1-9][0-9]*"
+
+if ! [[ "$VER" =~ ^v($NUMRE)\.($NUMRE)\.($NUMRE)?$ ]]; then
+	echo "malformed version: \"$VER\""
+	exit 1
+fi
```

Comment:
I would actually allow the pre-release stuff we did for SDK too. We didn't need it until we did, then we had to figure out a way to allow it. So I'd reuse what we had over there:
https://github.com/operator-framework/operator-sdk/blob/master/release.sh#L10-L18